src/H5system.c

```diff
@@ -732,8 +732,8 @@ H5_build_extpath(const char *name, char **extpath /*out*/)
 
             HDstrncpy(full_path, cwdpath, cwdlen + 1);
             if (!H5_CHECK_DELIMITER(cwdpath[cwdlen - 1]))
-                HDstrncat(full_path, H5_DIR_SEPS, path_len - (cwdlen + 1));
-            HDstrncat(full_path, new_name, path_len - (cwdlen + 1) - HDstrlen(H5_DIR_SEPS));
+                HDstrcat(full_path, H5_DIR_SEPS);
+            HDstrcat(full_path, new_name);
```

Comment:
How about a defensive rewrite using `snprintf`?

```
            int rc = snprintf(full_path, path_len, "%s%s%s", cwdpath,
                H5_CHECK_DELIMITER(cwdpath[cwdlen - 1]) ? "" : H5_DIR_SEPS,
                new_name);
            if (rc < 0 || (size_t)rc >= path_len) {
                HGOTO_ERROR(H5E_INTERNAL, H5E_CANTCOPY, FAIL,
                    "buffer would overflow")
            }
```

Reply:
I think that would certainly be a bit less prone to error, though maybe a little bit less clean. I'd be for doing that though.

Reply:
I'm all for using safer C string manip functions :) .

However, if there's an overflow, that is a logic error in this function as `path_len` is calcuated right above.